src/components/add-challenge/AddChallengeForm.tsx

```diff
@@ -0,0 +1,289 @@
+        // file_attachment: null
+    };
+
+    // hooks
+    const [formData, setFormData] = useState<CreateChallengeFormData>(defaultFormValues);
+
+    const [formErrorAlert, setFormErrorAlert] = useState<string | null>(null);
+
+    const toast = useToast();
+
+
+    // functions
+    const setErrorMessage = (errorMessage: string) => {
+        setFormErrorAlert(errorMessage);
+        window.scrollTo({
+            top: 0,
+            behavior: 'smooth',
+        });
+    };
+
+
+    const handleInputChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement> = (e) => {
+        const {name, value} = e.target;
+        setFormData({...formData, [name]: value});
+    }
+
+    const addChallengeEventHandler: React.FormEventHandler<HTMLFormElement> = (e) => {
+        const form = e.currentTarget;
+        e.preventDefault();
+
+        // validation
+        if (!form.checkValidity()){
+            form.reportValidity();
+            return;
+        }
+
+        const flag_is_formatted_properly = /^uactf\{.*\}$/.test(formData.flag);
+        if (!flag_is_formatted_properly){
+            setErrorMessage("Flag format should be in format: uactf{<flag>}");
+            return;
+        }
+
+        const map_division_form_value_to_backend_value: {
+            [key: string]: number[];
+        } = {
+            'all': [1,2],
+            '1': [1],
+            '2': [2]
+        }
+
+        const request_body: CreateChallengeRequest = {
+            ...formData,
+            creator_name: '', // TODO: put actual user name
+            division: map_division_form_value_to_backend_value[formData.division],
+            verified: false // TODO: this will automatically be true for super-admins and false for ua cd members
+        }
+        
+        createChallengeMutation.mutate(request_body);
+    }
+
+    const createChallenge = async (request_body: CreateChallengeRequest) => {
+        console.log(JSON.stringify(request_body))
+        const response = await fetch(
+            `${process.env.NEXT_PUBLIC_BACKEND_URL}/challenges/create`, 
+            {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(request_body),
+            }
+        );
+
+        return await response.json();
+    };
+
+    // mutation
+    const createChallengeMutation = useMutation({
+        mutationFn: createChallenge,
+        onSuccess: (data) => {
+            console.log('data:',data);
+            setFormErrorAlert(null);
+            setFormData(defaultFormValues);
+            toast({
+                title: 'Challenge created.',
+                status: 'success',
+                duration: 5000,
+                isClosable: true,
+            });
+        },
+        onError: (error) => {
+            console.log('error:' ,error);
+            setErrorMessage(error.message);
+        },
+    });
```

Comment:
Can we create a hooks folder and store this there? We want hooks to be separate so in case we can reuse them.